### ISSUE TITLE
Can the different child classes have different constructive methods with the same parent abstract class?

### ISSUE THREAD

**Issue by DerrickSaltFish** [nus-cs2030/2021-s1#142]
If there are more than one child classes that extend the same abstract class. Can they have different constructive methods?

**Comment by domlimm** [comment]
Is this what you mean?
```java
abstract class A {
    private final T field1;
    private final T field2;

    A(T field1, T field2) { ... }
}

class FirstChild extends A {
    private final T fieldFC;

    FirstChild (T field1, T field2) { super(field1, field2) }

    FirstChild(T field1, T field2, T fieldFC) { 
        super(field1, field2);
        this.fieldFC = fieldFC;
    }
}

class SecondChild extends A {
    private final T fieldSC;

    SecondChild (T field1, T field2) { super(field1, field2) }

    SecondChild(T field1, T field2, T fieldSC) { 
        super(field1, field2);
        this.fieldSC = fieldSC;
    }
}
```

**Comment by limeugene** [comment]
Yes the constructors for child classes can vary, but the child classes must have all the attributes that the abstract class has. (child class can add on to abstract class's attributes as well as seen in the above code "private final T fieldSC" etc)

**Comment by Terence-Sim** [comment]
Note that Constructors _**are never inherited by the child.**_ Nor can the child override the parent's constructor. This is true whether or not the parent class is abstract or concrete.

Instead, the child needs to provide its own Constructor (or else Java will provide a dummy one), which can have a different signature from the parent.

When a child is newly instantiated (by a call to one of its Constructor), if the child does not explicitly call super(), then Java implicitly calls it. That is, a parent instance is also created whenever a child instance is created. If the parent also has a parent, then this triggers the instantiation of that grandparent, and so on, all the way up to Object.
